From fb50b50e6fc94c713ea0710e236403bff1419a98 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 19 Jun 2018 10:34:06 +0530
Subject: [PATCH] Don't use the GCC ternary syntax extension

It breaks MSVC compatibility, and is only useful when the expression has
to be evaluated only once, which is not applicable here.
---
 tls/gnutls/gtlsclientconnection-gnutls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tls/gnutls/gtlsclientconnection-gnutls.c b/tls/gnutls/gtlsclientconnection-gnutls.c
index 97cb821..fae4b71 100644
--- a/tls/gnutls/gtlsclientconnection-gnutls.c
+++ b/tls/gnutls/gtlsclientconnection-gnutls.c
@@ -155,7 +155,7 @@ g_tls_client_connection_gnutls_compute_session_id (GTlsClientConnectionGnutls *g
           session_id = g_strdup_printf ("%s/%s/%d/%s", addrstr,
                                         server_hostname ? server_hostname : "",
                                         port,
-                                        cert_hash ?: "");
+                                        cert_hash ? cert_hash : "");
           gnutls->session_id = g_bytes_new_take (session_id, strlen (session_id));
           g_free (addrstr);
           g_free (cert_hash);
-- 
2.12.2.windows.2

